tests/integration/interface/umpire_fortran_interface.f

```diff
@@ -29,6 +29,17 @@ subroutine allocate_sixty_four
 
   end subroutine allocate_sixty_four
 
+  subroutine allocate_through_generic
+    integer(C_INT), pointer :: array(:)
+    type(UmpireAllocator) allocator
+    type(UmpireResourceManager) rm
+
+    rm = rm%get_instance()
+    allocator = rm%get_allocator_by_id(0)
+    call allocator%allocate_array( [ 10 ] , array)
+
```

Comment:
This looks pretty good.  I would flip the order of the parameters, if it's trivial.

```suggestion
allocator%allocate_array( array, [10] )
```
The FORTRAN devs are used to:
allocate(var(dim1,dim2,etc))
so when moving to your interface, I think having the pointer var first will feel more natural.